crates/networks/src/lib.rs

```diff
@@ -98,6 +104,10 @@ impl Networks {
     }
 
     pub async fn update_network(&mut self, old_name: &str, network: Network) -> Result<()> {
+        if self.networks.iter().any(|(_, n)| n.name == network.name) {
+            return Err(Error::AlreadyExists);
+        }
```

Comment:
won't this always return an error if you update a network without changing its name?
it will find itself in this loop

Reply:
Yes, it will, nice catch